packages/amplify-category-api/src/index.ts

```diff
@@ -205,3 +215,172 @@ export async function handleAmplifyEvent(context, args) {
   context.print.info(`${category} handleAmplifyEvent to be implemented`);
   context.print.info(`Received event args ${args}`);
 }
+
+type ApiResource = {
+  category: string;
+  resourceName: string;
+  githubInfo: {
+    path: string;
+    tokenSecretArn: string;
+  };
+  deploymentMechanism: DEPLOYMENT_MECHANISM;
```

Comment:
The problem with that is, when we are generating the stack using cdk we need that value in order to configure the stage action on the pipeline otherwise it will be a more complex solution because we would have to add cfn conditions.